spi/control-plane/contract-spi/src/testFixtures/java/org/eclipse/dataspaceconnector/contract/offer/store/ContractDefinitionStoreTestBase.java

```diff
@@ -0,0 +1,464 @@
+/*
+ *  Copyright (c) 2022 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial Tests
+ *       Microsoft Corporation - Method signature change
+ *       Microsoft Corporation - refactoring
+ *       Microsoft Corporation - added tests
+ *       Fraunhofer Institute for Software and Systems Engineering - added tests
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - improvements
+ *
+ */
+
+package org.eclipse.dataspaceconnector.contract.offer.store;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.spi.contract.offer.store.ContractDefinitionStore;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.query.SortOrder;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.eclipse.dataspaceconnector.contract.offer.store.TestFunctions.createContractDefinition;
+import static org.eclipse.dataspaceconnector.contract.offer.store.TestFunctions.createContractDefinitions;
+import static org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression.SELECT_ALL;
+
+public abstract class ContractDefinitionStoreTestBase {
+
+    public ContractDefinitionStoreTestBase() {
+        System.setProperty("contractdefinitionstore.supports.nestedfields", String.valueOf(supportNestedFields()));
```

Comment:
Here, the situation is, that the Cosmos impl uses a cache, which in turn uses utility that uses Reflection to construct the `Predicate` for matching items to the query. That utility cannot (yet) match collection types, only scalars. Thus, it is not about nested fields here, but rather about collection types. Could you adapt the property name accordingly?

I'll create another issue to follow up on this.